docs/stateful_models.md

```diff
@@ -0,0 +1,317 @@
+
+
+Stateful model can either be subject to idle sequence cleanup or not.
+You can set this **per model** with `idle_sequence_cleanup` parameter. 
+If set to `true` sequence cleaner will check that model. Otherwise sequence cleaner will ommit that model and its inactive sequences will not get removed. By default this value is set to `true`.
+
+
+## Run inference on stateful model <a name="stateful_inference"></a>
+
+### Special inputs for sequence handling <a name="stateful_inputs"></a>
+
+Stateful model works on consecutive inference requests that are associated with each other and form **sequence** of requests. Single stateful model can handle multiple sequences at a time. When model server receives requests for stateful model it needs to know which request belongs to which sequence. OVMS also needs to know where is the begging and the end of the sequence to properly manage its resources.
+
+To provide that information, requests to stateful models must contain additional inputs: 
+- `sequence_id` - which is 64-bit unsigned integer identifying the sequence (unique in the scope of the model instance). Value 0 is equivalent to not providing this input at all.
+- `sequence_control_input` - which is 32-bit unsigned integer indicating sequence start and end. Accepted values are: 
+   - 0 - no control input (has no effect - equivalent to not providing this input at all)
+   - 1 - indicates start of the sequence
+   - 2 - indicates end of the sequence
+
+**Note**: Model server also appends `sequence_id` to every response.
+
+**Both `sequence_id` and `sequence_control_input` must be provided as 1-dimensional arrays of size 1.**  
+_See examples for gRPC and HTTP below_.
+
+In order to successfully infer the sequence, perform these actions:
+1. **Send the first request in the sequence and signal sequence start.**
+
+   To start the sequence you need to add `sequence_control_input` with value of 1 to your request's inputs. You can also:
+      - add `sequence_id` with the value of your choice or
+      - add `sequence_id` with 0 or do not add `sequence_id` at all - in this case model server will provide unique id for the sequence and since it'll be appended to the outputs, you'll be able to read it and use with the next requests. 
+
+2. **Send remaining requests except the last one.**
+
+   To send requests in the middle of the sequence you need to add `sequence_id` of your sequence. In this case `sequence_id` is mandatory and not providing this input or setting it's value to 0 is not allowed.
+
+3. **Send the last request in the sequence and signal sequence end.**
+
+   To end the sequence you need to add `sequence_control_input` with the value of 2 to your request's inputs. You also need to add `sequence_id` of your sequence. In this case `sequence_id` is mandatory and not providing this input or setting it's value to 0 is not allowed.
+
```

Comment:
Add links to exemplary clients.

Reply:
Both available clients have been linked under **Inference via gRPC** and **Inference via HTTP** paragraphs after additional requirements (inputs format) are described and after user sees code snippets with examples of how to add stateful specific data to the requests depending on the interface (grpc or http). 
Is it worth to link examples also here?